sunpy/map/mapcubed.py

```diff
@@ -0,0 +1,590 @@
+
+    def _derotate(self):
+        """Derotates the layers in the MapCube"""
+        pass
+
+    @property
+    def instrument(self):
+        """Instrument name"""
+        return self._meta[self.ref_index].get('instrume', "").replace("_", " ")
+
+    @property
+    def measurement(self):
+        """Measurement name, defaults to the wavelength of image"""
+        return u.Quantity(self._meta[self.ref_index].get('wavelnth', 0), self._meta[self.ref_index].get('waveunit', ""))
+
+    @property
+    def wavelength(self):
+        """wavelength of the observation"""
+        return u.Quantity(self._meta[self.ref_index].get('wavelnth', 0), self._meta[self.ref_index].get('waveunit', ""))
+
+    @property
+    def observatory(self):
+        """Observatory or Telescope name"""
+        return self._meta[self.ref_index].get('obsrvtry', self._meta[self.ref_index].get('telescop', "")).replace("_", " ")
+
+    @property
+    def detector(self):
+        """Detector name"""
+        return self._meta[self.ref_index].get('detector', "")
+
+    @property
+    def dimensions(self):
+        """
+        The dimensions of the array (x axis first, y axis second).
+        """
+        return self.data.shape[1], self.data.shape[0]
+
+    @property
+    def dtype(self):
+        """
+        The `numpy.dtype` of the array of the map.
+        """
+        return self.data.dtype
+
+    @property
+    def date(self):
+        """Observation time"""
+        return [parse_time(m.get('date-obs', 'now')) for m in self._meta]
+
+    @property
+    def scale(self):
+        """
+        Image scale along the x and y axes in units/pixel (i.e. cdelt1,
+        cdelt2)
+        """
+        #TODO: Fix this if only CDi_j matrix is provided
+        return self._get_map(self.ref_index).scale
+
+    @property
+    def units(self):
+        """
+        Image coordinate units along the x and y axes (i.e. cunit1,
+        cunit2).
+        """
+        return self._get_map(self.ref_index).units
+
+    @property
+    def exposure_time(self):
+        """Exposure time of the image in seconds."""
+        return self._meta[self.ref_index]['exptime'] * u.s
+
+    @property
+    def rotation_matrix(self):
+        return self._get_map(self.ref_index).rotation_matrix
+
+    def meta(self, key):
+        """The Meta."""
+        result = [meta[key] for meta in self._meta]
+        # check to see if they are all the same if so just return one value
+        if np.all([r == result[0] for r in result]):
+            result = [result[0]]
+        return result
+
+    def submap(self, range_a, range_b, range_c=None):
+        """Returns a submap of the map with the specified range.
+        """
+        new_maps = []
+        for i in range(0, len(self)):
+            new_maps.append(self._get_map(i).submap(range_a, range_b))
+        return MapCubed(new_maps)
+
+    def lightcurve(self, location_a, location_b, range_c=None):
```

Comment:
The API here should be fixed. Would likely be better and more intuitive if a light curve were output with something like `mycubed[:,x,y]`